graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/CreateTableFetcher.java

```diff
@@ -23,21 +23,35 @@
 import com.datastax.oss.driver.api.querybuilder.schema.CreateTableWithOptions;
 import graphql.schema.DataFetchingEnvironment;
 import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.AuthorizationService;
+import io.stargate.auth.Scope;
+import io.stargate.auth.UnauthorizedException;
 import io.stargate.db.Persistence;
+import io.stargate.graphql.web.HttpAwareContext;
 import java.util.List;
 import java.util.Map;
 
 public class CreateTableFetcher extends DdlQueryFetcher {
 
-  public CreateTableFetcher(Persistence persistence, AuthenticationService authenticationService) {
-    super(persistence, authenticationService);
+  public CreateTableFetcher(
+      Persistence persistence,
+      AuthenticationService authenticationService,
+      AuthorizationService authorizationService) {
+    super(persistence, authenticationService, authorizationService);
   }
 
-  public String getQuery(DataFetchingEnvironment dataFetchingEnvironment) {
+  public String getQuery(DataFetchingEnvironment dataFetchingEnvironment)
+      throws UnauthorizedException {
+    String keyspaceName = dataFetchingEnvironment.getArgument("keyspaceName");
+    String tableName = dataFetchingEnvironment.getArgument("tableName");
+
+    HttpAwareContext httpAwareContext = dataFetchingEnvironment.getContext();
+    String token = httpAwareContext.getAuthToken();
+    authorizationService.authorizeSchemaWrite(token, keyspaceName, tableName, Scope.CREATE);
```

Comment:
These 5 lines are the same for all table operations, only the scope changes. We could extract a `TableFetcher` parent class where `getQuery` performs the check and delegates to an abstract `getQuery(DataFetchingEnvironment, String, String)`.

Reply:
Made the change. Let me know what you think.